pkg/controller/direct/logging/logmetric_controller.go

```diff
@@ -224,66 +222,122 @@ func logMetricStatusToKRM(in *api.LogMetric, out *krm.LoggingLogMetricStatus) er
 }
 
 func (a *logMetricAdapter) Update(ctx context.Context, u *unstructured.Unstructured) error {
-	update := new(api.LogMetric)
-	*update = *a.actual
-
-	if ValueOf(a.desired.Spec.Description) != a.actual.Description {
-		update.Description = ValueOf(a.desired.Spec.Description)
-	}
-	if ValueOf(a.desired.Spec.Disabled) != a.actual.Disabled {
-		update.Disabled = ValueOf(a.desired.Spec.Disabled)
-	}
-	if a.desired.Spec.Filter != a.actual.Filter {
-		// todo acpana: revisit UX, err out if filter of desired is empty
-		if a.desired.Spec.Filter != "" {
-			update.Filter = a.desired.Spec.Filter
-		} else {
-			// filter is a REQUIRED field
-			update.Filter = a.actual.Filter
+	latest := a.actual
+
+	if a.hasChanges(ctx, u) {
+		update := new(api.LogMetric)
+		*update = *a.actual
+
+		if ValueOf(a.desired.Spec.Description) != a.actual.Description {
+			update.Description = ValueOf(a.desired.Spec.Description)
+		}
+		if ValueOf(a.desired.Spec.Disabled) != a.actual.Disabled {
+			update.Disabled = ValueOf(a.desired.Spec.Disabled)
+		}
+		if a.desired.Spec.Filter != a.actual.Filter {
+			// todo acpana: revisit UX, err out if filter of desired is empty
+			if a.desired.Spec.Filter != "" {
+				update.Filter = a.desired.Spec.Filter
```

Comment:
+1 to err out if the spec.filter is unset. We shall not allow unspecified required fields.

Reply:
I don't think I touched this code here other than indentation.  Maybe an AI for @acpana  ?

Reply:
https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/1798

I don't think it's a blocker for 1.118 😛  but I'll leave that up to y'all